the-blue-alliance-ios/Core Data/Event/Event.swift

```diff
@@ -448,6 +448,17 @@ extension Event: Locatable, Surfable, Managed {
         }
     }
 
+    /**
+     Same as title but return nil if title isn't in Week 0.5..7
+     */
+    public var weekString: String? {
+        if title.contains("Week") {
```

Comment:
We shouldn't need to do any substring matching here - we have the logic already to calculate a week string based on Event properties. We can pull it out of the previous `weekString` method and use it here